### DISCUSSION TITLE
First Sun image

### DISCUSSION THREAD

**Discussion by Victoria-Samboco** [ratt-ru/solarkat#7] [General]
Good news :) Finally I have the first image of the Sun. It is a bit distorted I don't know why but is there. This are images of the same scan (scan1- with 5min observation), were in the left is the MFS-image.fits and at the right is the MFS-dirty.fits images. 

This is the command I used to construct the image:

wsclean -name scans1 -data-column CORRECTED_DATA -size 10240 10240 -channels-out 8  -niter 0  -mgain 0.9  -weight briggs -1.0 -scale 1.1asec -join-channels  -shift 22h10m20.8439s   -11d17m19.604s  -subtract-model -auto-threshold 1 -no-update-model-required -auto-mask 5 1645272078_sdp_l0_1024ch_J0333-2741_scan1.ms

![scan1_sun](https://user-images.githubusercontent.com/95472901/182851760-30ebc90d-0e7e-4f4d-9f66-282bda6d63d4.png)
![scan1](https://user-images.githubusercontent.com/95472901/182851818-b1a71a34-7a24-4666-9dfc-eabcffbccaf9.png)
![scan1](https://user-images.githubusercontent.com/95472901/182851916-5bd5e2e5-bbe8-4802-9be4-10f9a3a3a8ba.png)

**Comment by Victoria-Samboco** [comment]
Scan7
![scan7](https://user-images.githubusercontent.com/95472901/182957635-6522732c-b965-45bb-83c8-b86549a717a2.png)

**Reply by o-smirnov** [reply]
Looking good! You can find SDO images from the Sun for that date on https://www.nasa.gov/mission_pages/sdo/the-sun-now/index.html, see if the sunspots match up.

**Reply by IanHeywood** [reply]
https://solarmonitor.org/

This is also a good resource for daily Sun images. The 174 Angstrom image usually shows good correspondence with the radio.

**Comment by IanHeywood** [comment]
Congratulations, you are officially a solar radio astronomer. Now we just have to make sense of it all. :)

The distortion in the shape is strange. I have seen distorted images of the sun occuring as it sets, but the reason I asked for a scan 7 image is because I have an image of the sun for comparison, but in my image the sun retains its expected shape. I think the only difference is that I used `chgcentre` and did not know about the `shift` parameter.

I would expect both of these operations to perform in the same way as they are written by the same person to do the same function, however we should rule out or confirm whether it is the `shift` parameter that is causing the distortion. If it is then we can revert to using `chgcentre` (and send an email to André reporting this behaviour).

Can you please perform this test with the scan 7 MS? I would suggest the following:

- Duplicate the scan 7 Measurement Set 
- Run `chgcentre` on it with the RA and Dec appropriate for scan 7
- Repeat your `wsclean` command above on the shifted MS but omit the `shift` parameter

**Comment by Victoria-Samboco** [comment]
Good . Iam Happy to hear. Thanks you :)
I will perform the experiment whit the change centre and see what happen.

**Comment by Victoria-Samboco** [comment]
Hi all,  more good news.
So,  above we have the image from scan7 using chgcentre command . And the second  is the comparison between the image of the Sun (also Scan7) with chgcentre (left) and the other using  -shift command in wsclean. Here the difference is notable,  for some reason the shift command seems that causes distortions in the sources. I really would like to know what exacty is this command doing to cause this distortions. Well however the chgcentre command is working better than the shift command, at least in the presentation of the image.

![Sun-with-chgcentre](https://user-images.githubusercontent.com/95472901/183215595-e5bc56c2-e110-4618-9a1b-c9c0d85cad51.png)
![Sun_scan7](https://user-images.githubusercontent.com/95472901/183215648-13eaa27a-4ed9-4683-a88f-0903e2a4b838.png)

**Reply by IanHeywood** [reply]
That looks better! I think we should retain the use of `chgcentre` instead of using `-shift` then. We should find out from André whether this is expected behaviour or a bug in wsclean.

**Comment by o-smirnov** [comment]
Now that's more like it! Great stuff.

I can ping Andre with the image to see what's up. I'm sure he'll want to fix it.

**Comment by o-smirnov** [comment]
Actually we misunderstand ``-shift`` I reckon: https://wsclean.readthedocs.io/en/latest/w_snapshot_algorithm.html?highlight=shift#w-snapshot-algorithm

> The chgcentre command rotates the coordinate system whereas WSClean’s -shift parameter shifts the coordinate system along the tangent plane. The result is that imaging is done in a different projection that is still centered on the target location, but that has smaller w-terms. The [chgcentre page](https://wsclean.readthedocs.io/en/latest/chgcentre.html) provides further information about the chgcentre tool.

**Comment by IanHeywood** [comment]
Ah-ha! My bad, I just read the command line description and didn't realise it wasn't doing what we wanted. Sorry!
